Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -97,54 +96,106 @@ class RpcapdServerInitializer
 {
 private:
 	HANDLE m_ProcessHandle;
+	HANDLE m_JobHandle;
 
-public:
+	void killProcessAndCloseHandles()
+	{
+		if (m_ProcessHandle != nullptr)
+		{
+			TerminateProcess(m_ProcessHandle, 0);
+			CloseHandle(m_ProcessHandle);
+			m_ProcessHandle = nullptr;
+		}
 
-	RpcapdServerInitializer(bool activateRemoteDevice, const std::string &ip, uint16_t port) : m_ProcessHandle(nullptr)
+		if (m_JobHandle != nullptr)
+		{
+			CloseHandle(m_JobHandle);
+			m_JobHandle = nullptr;
+		}
+	}
+public:
+	RpcapdServerInitializer(bool activateRemoteDevice, const std::string& ip, uint16_t port)
+		: m_ProcessHandle(nullptr), m_JobHandle(nullptr)
 	{
 		if (!activateRemoteDevice)
 			return;
 
 		std::string cmd = "rpcapd\\rpcapd.exe";
-		std::ostringstream args;
-		args << "rpcapd\\rpcapd.exe -b " << ip << " -p " << port << " -n";
+		std::string args;
```

Comment:
As you already count the size of usage. recommend to use `std::snprintf`

Reply:
What is the benefit of `std::snprintf`?

Reply:
we can simplify the code as:
```
    // Define a buffer large enough to hold the formatted string
    char buffer[256];

    // Use snprintf to format the string
    std::snprintf(buffer, sizeof(buffer), "rpcapd\\rpcapd.exe -b %s -p %d -n", ip.c_str(), port);
```

Reply:
well, writing it out, it ended up closer to this.
```
std::array<char, 256> args;
int res = std::snprintf(args.data(), args.size(), "rpcapd\\rpcapd.exe -b %s -p %d -n", ip.c_str(), port);
if (res < 0)
{
	throw std::runtime_error("Error during string formatting");
}
else if (res >= args.size())
{
	// We should never get here with reasonable input but you never know.
	throw std::runtime_error("Buffer overflow. Are you sure the IP is a valid string?");
}
```

Reply:
Implemented a8316a9